### PR TITLE
Fix introspecting external @Entity classes

### PR DESCRIPTION
**Pull request by dstepanov**
It's not enough to check if the class has `@Introspected`, external classes present the annotation because if the mapper.

Not sure if the fix is correct.

### CODE REVIEW

File: core/src/main/java/io/micronaut/core/io/service/SoftServiceLoader.java [hunk 1]
```diff
@@ -197,23 +198,20 @@ public boolean hasNext() {
...
+                        .map(line -> {
+                            int i = line.indexOf('#');
+                            if (i > -1) {
+                                line = line.substring(0, i);
+                            }
+                            return line;
+                        })
+                        .collect(Collectors.toList());
+                    unprocessed = lines.iterator();
+                } catch (IOException | UncheckedIOException e) {
```
Comment (dstepanov) [inline core/src/main/java/io/micronaut/core/io/service/SoftServiceLoader.java]: Not related, somehow I would get `UncheckedIOException` a lot during Gradle invocation

**Comment by graemerocher** [comment]
Yeah this seems like a hack, must be a better a way 🤔

**Comment by jameskleeh** [comment]
@dstepanov Can you more clearly describe the problem?

**Comment by dstepanov** [comment]
@jameskleeh The problem here is that the mapper adds `@Introspected` to all `@Entity` but external classes are not processed. That causes `@Introspected(classes = ValuesEntity.class)` to skip the class because of the founded `@Introspected` annotation.

**Comment by graemerocher** [comment]
seems the correct solution would be to make sure mappers are applied to external classes as well?

**Comment by dstepanov** [comment]
Mapper is applied correctly but `IntrospectedTypeElementVisitor` doesn't process classes in a jar.
Manually defining `classes`/`package` skips the class because annotation is present because of the mapper.

**Comment by graemerocher** [comment]
There are only tests for Groovy code in this PR, the change needs to add tests for Java as well

**Comment by dstepanov** [comment]
@graemerocher Added

**Comment by dstepanov** [comment]
@graemerocher Should this be merged?

**Comment by graemerocher** [comment]
Thanks for the contribution!

**Pull request merged**